### DISCUSSION TITLE
User registration

### DISCUSSION THREAD

**Discussion by Aloe-recite** [CorentinTh/enclosed#358] [Ideas]
Hi, I'm aware that it's not an easy task, but it would be nice considering the addition of a graphical way for users to register and having to login to use the app. Currently I protect it with Cloudflare Zero Trust which I find quicker and easier than hashing email and passwords as specified in the docs, but I still need to manually add the email of my friends, aske them, and enable many different auth providers since not everyone has a Gmail account. A registration-login page would make it far easier and strike a good balance between an open instance (much undesirable in my use case) and a too-closed solution. HAving the ability to restrict registration to specific email domains would be the cherry on top. Thanks :)

**Comment by CorentinTh** [comment]
Hi @Aloe-recite, thanks for the feedback and suggestions :pray: 

It's currently something I plan to do, use a proper persisted storage solution for users rather than relying on env variables
In current state of Enclosed I think the easiest way to implent this would be to use [namespaced unstorage instance](https://unstorage.unjs.io/guide/utils#namespace), one for notes, one for users, so we'll have the same persistence layer as currently and migrating to the future version will be easier

It'll allow simpler user registration and the posibility to add stuff like 2FA/MFA

A point of attention will be to securely have retro-compatibility with previously created notes
